src/BuiltInTools/dotnet-watch/LaunchBrowserFilter.cs

```diff
@@ -75,7 +75,17 @@ public async ValueTask ProcessAsync(DotNetWatchContext context, CancellationToke
 
                         var pathToMiddleware = Path.Combine(AppContext.BaseDirectory, "middleware", "Microsoft.AspNetCore.Watch.BrowserRefresh.dll");
                         context.ProcessSpec.EnvironmentVariables["DOTNET_STARTUP_HOOKS"] = pathToMiddleware;
-                        context.ProcessSpec.EnvironmentVariables["ASPNETCORE_HOSTINGSTARTUPASSEMBLIES"] = "Microsoft.AspNetCore.Watch.BrowserRefresh";
+
+                        const string hostingStartupAssembliesName = "ASPNETCORE_HOSTINGSTARTUPASSEMBLIES";
+                        var existingHostingStartupAssemblies = Environment.GetEnvironmentVariable(hostingStartupAssembliesName);
+                        if (string.IsNullOrEmpty(existingHostingStartupAssemblies))
+                        {
+                            context.ProcessSpec.EnvironmentVariables[hostingStartupAssembliesName] = $"{existingHostingStartupAssemblies};Microsoft.AspNetCore.Watch.BrowserRefresh";
```

Comment:
This seems counter-intuitive to me. I believe we want to set `existingHostingStartupAssemblies` in the environment variable when it is _not_ null or empty?

Reply:
That's what happens when I modify this PR in the browser <_<